model-optimizer/mo/middle/passes/fusing/fuse_linear_ops.py

```diff
@@ -84,6 +91,36 @@ def _fuse_mul(graph: Graph, node: Node, fuse_nodes: list, backward: bool = True)
         ch_dim = weights_port.data.get_attr('output_channel_dim' if backward else 'input_channel_dim')
         shape = np.array([weights_port.data.get_shape()[ch_dim]])
 
+        group = fuse_node.soft_get('group', 1)
+        if group != 1 and shape[0] % group != 0:
+            shape *= group
+            # reshape weights [I,O,H,W] -> [I/G,G*O,H,W]
+            weights_node = weights_port.get_connection().get_source().node
+            weights_name = weights_node.soft_get('name', weights_node.id)
+
+            shape_node = Shape(graph, {'name': weights_name + '/Shape'}).create_node()
+            shape_node.in_port(0).connect(weights_node.out_port(0))
+
+            reshape_before = Reshape(graph, {'name': weights_name + '/ReshapeBefore',
+                                             'override_output_shape': True}).create_node()
+            mul_const = float_array([1. / group, 1 * group])
+            for x in range(wdims_number - 2):
+                mul_const = np.append(mul_const, 1.)
+            mul = create_op_with_const_inputs(graph, Mul, {1: mul_const}, {'name': weights_name + '/Mul'})
+            mul.in_port(0).connect(shape_node.out_port(0))
+            mul.out_port(0).connect(reshape_before.in_port(1))
+
+            cast_shape = Cast(graph, {'dst_type': np.int64}).create_node()
+            reshape_before.in_port(1).get_connection().insert_node(cast_shape)
+
+            reshape_after = Reshape(graph, {'name': weights_name + '/ReshapeAfter'}).create_node()
+            reshape_after.in_port(1).connect(shape_node.out_port(0))
+            rename_nodes([(weights_node, weights_name + '/Weights'), (reshape_after, weights_name)])
+
+            weights_port.get_connection().insert_node(reshape_before)
+            weights_port.get_connection().insert_node(reshape_after)
+            weights_port = reshape_after.in_port(0)
```

Comment:
General comment. We have a transformation that modifies weights of group convolutions at runtime.

@ilyachur , @GlebKazantaev how much will be the performance impact for the first inference time?